### PR TITLE
Bundling and minification package

### PR DESCRIPTION
**Pull request by vadym-hyryn**
[User story 27467](https://dev.azure.com/orckestra001/OrckestraCommerce/_workitems/edit/27467/)
Orckestra.Web.Css.Less and Orckestra.Web.Css.Sass where updated to use DI with Composite ConfigureServices method instead of manual static class. Also some fixings and improvements there.
I have tested BundlingAndMinification package productivity with google audits, and compared it with previous version, so in average there are no difference in performance. But added different improvements. Additional info in Install.xls file :)

### CODE REVIEW

**Comment by napernik** [comment]
It makes sense to use Composite.Core.WebClient.Renderings.Page.IPageContentFilter.IPageContentFilter interface instead of a response filter. This way you wouldn't have to handle gzip encoding, as well as achieve much better memory usage performance, as with it you can process the page output, when it is already represented as an XhtmlDocument object, and there's no need to parse a string, and convert the document back into string.
All that's needed is registering an implementation of the interface on startup.

**Comment by vadym-hyryn** [comment]
> It makes sense to use Composite.Core.WebClient.Renderings.Page.IPageContentFilter.IPageContentFilter interface instead of a response filter. This way you wouldn't have to handle gzip encoding, as well as achieve much better memory usage performance, as with it you can process the page output, when it is already represented as an XhtmlDocument object, and there's no need to parse a string, and convert the document back into string.
> All that's needed is registering an implementation of the interface on startup.

Fully agreed, this will be better, thank you about noticing about this C1 opportunity :) I have updated the pull request

**Comment by vadym-hyryn** [comment]
> The solution folder name is "Composite.Web.BundlingAndMinification", but the solution name is "Orckestra.Web.BundlingAndMinification", which makes harder to find sources by the package name within this repository.

Renamed a folder

File: Composite.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/PageContentFilter.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+        //to be executed in the end, after all another filters
+        public int Order => int.MaxValue;
+
+        public void Filter(XhtmlDocument document, IPage page)
+        {
+            if (string.IsNullOrEmpty(document?.Root?.Document?.ToString()))
+            {
+                return;
+            }
```
[the remark this answers is not in the document]
Comment (vadym-hyryn) [reply Composite.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/PageContentFilter.cs]: Deleted this check

File: Composite.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/PageContentFilter.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            XDocumentType doctype = document.Document.DocumentType;
+            if (doctype?.InternalSubset?.Trim() == string.Empty)
+            {
+                doctype.InternalSubset = null;
+            }
+
+            HttpContext httpContext = HttpContext.Current;
+            if ((!_bundleAndMinifyScripts && !_bundleAndMinifyStyles) ||
+                httpContext.IsDebuggingEnabled ||
+                !(httpContext.Handler is Page) ||
```
[the remark this answers is not in the document]
Comment (vadym-hyryn) [reply Composite.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/PageContentFilter.cs]: Thanks, deleted page handler type check

File: Composite.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/PageContentFilter.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                IsAdminConsoleRequest(httpContext))
+            {
+                return;
+            }
+
+            XhtmlDocument optimizedPage = GetOptimizedDocument(document);
+            if (optimizedPage?.Root != null)
+            {
+                document.Root.ReplaceWith(optimizedPage.Root);
+            }
```
[the remark this answers is not in the document]
Comment (vadym-hyryn) [reply Composite.Web.BundlingAndMinification/Orckestra.Web.BundlingAndMinification/PageContentFilter.cs]: Rewrited the code, also provided perfomance tests with ?c1mode=perf to main C1 page, it seems okay now

Time processing results (ms)

  | without package | with prev. 1.3.2   version of package | with new package   before update | with new package after current update
-- | -- | -- | -- | --
  | 14.5 | 14.6 | 13.5 | 13.7
  | 25.1 | 14.9 | 14.6 | 12.6
  | 14.2 | 13.3 | 13.7 | 15.9
  | 13.8 | 16.2 | 11.9 | 12.7
  | 14.3 | 16 | 19.3 | 14.2
  | 15.7 | 14 | 12 | 13.8
  | 15.5 | 14.5 | 15 | 13.8
  | 11.9 | 14.3 | 12.7 | 13.3
  | 14.1 | 16.2 | 13.5 | 15.1
  | 14 | 18.9 | 14.1 | 14.2
  | 15.8 | 14.3 | 14.9 | 13.7
  | 14.5 | 14.3 | 13.5 | 11.6
  | 15 | 18.1 | 11.7 | 13.2
  | 15.8 | 14.7 | 15.8 | 15.2
  | 14.4 | 15.2 | 14.5 | 14.3
  | 14.5 | 15.6 | 14.3 | 15.1
  | 15.4 | 16.3 | 13.2 | 12.2
  | 14.7 | 15.6 | 14.7 | 21.5
  | 13.3 | 15.1 | 14.8 | 12.1
Mediana | 14.5 | 15.1 | 14.1 | 13.8
Difference | x | 0.6 | -0.4 | -0.7

Memory processing results (kb)

  | without package| with prev. 1.3.2   version of package | with new package   before update| with new package after current update
-- | -- | -- | -- | --
  | 4807 | 5167 | 4978 | 4768
  | 4944 | 5183 | 4985 | 4760
  | 4761 | 5167 | 4981 | 4760
  | 4764 | 5167 | 4980 | 4760
  | 4759 | 5166 | 5135 | 4763
  | 4760 | 5167 | 4981 | 4760
  | 4770 | 5174 | 4978 | 4760
  | 4760 | 5167 | 4981 | 4766
  | 4765 | 5174 | 4981 | 4768
  | 4761 | 5463 | 4979 | 4763
  | 4776 | 5167 | 4991 | 4768
  | 4787 | 5171 | 5015 | 4771
  | 4768 | 5178 | 4981 | 4776
  | 4771 | 5167 | 4974 | 4776
  | 4768 | 5170 | 4973 | 4779
  | 4768 | 5175 | 4973 | 4760
  | 4777 | 5167 | 4975 | 4758
  | 4768 | 5163 | 4973 | 4761
  | 4773 | 5163 | 4975 | 4760
Mediana | 4768 | 5167 | 4980 | 4763
Difference | x | 399 | 212 | -5

**Comment by vadym-hyryn** [comment]
Thank you for a very detailed and informal comments :)

**Pull request closed without merging**